src/main/java/frc/robot/subsystems/vision/VisionEstimation.java

```diff
@@ -157,26 +173,29 @@ private Matrix<N3, N1> confidenceCalculator(EstimatedRobotPose estimation) {
                         / (1
                                 + ((estimation.targetsUsed.size() - 1) * VisionConstants.TAG_PRESENCE_WEIGHT)));
 
+        confidenceTelemetry.addNumber(confidenceMultiplier);
         return visionMeasurementStdDevs.times(confidenceMultiplier);
     }
 
-    public void estimatorChecker(Vision estimator, CountPerPeriodTelemetry getAtomicCountTelemetry) {
+    public void estimatorChecker(Vision estimator, CountPerPeriodTelemetry getAtomicCountTelemetry,
+                    MeanPerPeriodTelemetry confidenceTelemetry) {
         var cameraPose = estimator.grabLatestEstimatedPose();
 
         if (cameraPose != null) {
-            SmartDashboard.putString("camerapose", cameraPose.estimatedPose.toString());
             // New pose from vision
             var pose2d = cameraPose.estimatedPose.toPose2d();
             if (originPosition == kRedAllianceWallRightSide) {
                 pose2d = flipAlliance(pose2d);
             }
 
-            swerveSubsystem.addVisionReading(pose2d, cameraPose.timestampSeconds,
-                    confidenceCalculator(cameraPose));
+            final var confidence = confidenceCalculator(cameraPose, confidenceTelemetry);
+            swerveSubsystem.addVisionReading(pose2d, cameraPose.timestampSeconds, confidence);
 
             // Update "get atomic count" telemetry
             getAtomicCountTelemetry.incCount(1);
         }
-    }
 
+        final String smartDashboardCamPoseKey = "Vision/" + estimator.cameraName + " - last pose";
+        SmartDashboard.putString(smartDashboardCamPoseKey, cameraPose != null ? cameraPose.estimatedPose.toString() : "no pose");
```

Comment:
Reporting the pose from each camera separately so we can see what each camera thinks.